### PR TITLE
Refactor: organize loops file into loops module (Delegated)

### PR DESCRIPTION
**Pull request by Y-Nak**
@flip1995 @nahuakang 

closes #6693
r? @flip1995 

 As we talked about in the PM of Zulip,  this PR is a delegated PR from @nahuakang.

Changes from the last commit of #6693:
1. Unify the name of the main entries of all modules to check, that was pointed out [here](https://github.com/rust-lang/rust-clippy/pull/6693#discussion_r582897713)
2. Simplify ` check_for_loop_arg`, that was pointed out [here](https://github.com/rust-lang/rust-clippy/pull/6693#discussion_r582914044) and [here](https://github.com/rust-lang/rust-clippy/pull/6693#discussion_r582929910)
3. Resolve conflicts

changelog: Refactor `loops.rs` file into `loops` module.

### CODE REVIEW

**Comment by Y-Nak** [comment]
Changes: Resolve conflicts.

File: clippy_lints/src/loops/for_loop_arg.rs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        "to write this more concisely, try",
+        format!("&{}{}", muta, object),
+        applicability,
+    )
+}
+
+/// Returns `true` if the type of expr is one that provides `IntoIterator` impls
+/// for `&T` and `&mut T`, such as `Vec`.
+#[rustfmt::skip]
+fn is_ref_iterable_type(cx: &LateContext<'_>, e: &Expr<'_>) -> bool {
```
Comment (flip1995) [inline clippy_lints/src/loops/for_loop_arg.rs:126]: This function already exists in `explicit_iter_loop.rs`

**Review by flip1995 — COMMENTED** [review]
Found something to improve. I will continue reviewing this ASAP (I'm half way through)

File: clippy_lints/src/loops/for_loop_arg.rs (hunk 1 above)
Comment (Y-Nak) [reply clippy_lints/src/loops/for_loop_arg.rs:126]: `for_loop_arg.rs` seems to be no longer used, I mistakenly put it in there while rebasing, I'll just remove it.

**Comment by flip1995** [comment]
LGTM. I haven't looked in-depth on all the visitors, but I have the feeling that it might be worth looking into sharing those between the lints. Not su if that is possible though.

@bors r+

Thanks for the initial work @nahuakang and thanks for picking it up @Y-Nak !

**Pull request merged**